src/resolvers/resolvers.ts

```diff
@@ -8,32 +8,50 @@ import {
   findUserById,
   findUserData,
   findUserEmail,
-  generateToken,
   toHashPassword,
-  validateEmail
+  validateEmail,
+  getUsers
 } from '../functions';
 import { getUserIdByToken } from '../utils/get-userId-by-token';
+import { generateToken } from '../utils/generate-token';
 
 export const resolvers = {
   Query: {
     user: async (_, args, context: { req }) => {
       const userId = getUserIdByToken(context);
-
-      if (userId !== args.id) {
+      if (!userId) {
         throw new CustomError('Invalid token', 401);
       }
       const user = await findUserById(args.id);
       if (!user) {
         throw new CustomError('User not found.', 404);
       }
-
       return user;
+    },
+    users: async (_, args, context: { req }) => {
+      const userId = getUserIdByToken(context);
+      if (!userId) {
+        throw new CustomError('Invalid token', 401);
+      }
+      const quantity: number = args.quantity;
+      const page: number = args.page;
+      const users = await getUsers(quantity, page);
```

Comment:
Olhando esse código, dá impressão que o retorno do `getUsers` é um array de usuários, tanto pelo nome da variável quanto pelo nome da função. Uma coisa que vc pode fazer aqui é usar destruct pra assinalar 2 variáveis com nomes melhores. Seria legal tbm deixar o nome da função mais claro no sentido de listar e contar. Por exemplo:


```suggestion
      const [users, count] = await listAndCountUsers(quantity, page);
```